hazelcast-client-legacy/src/main/java/com/hazelcast/client/spi/impl/CallIdSequence.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.spi.impl;
+
+
+import com.hazelcast.core.HazelcastOverloadException;
+
+import java.util.concurrent.atomic.AtomicLongArray;
+
+import static com.hazelcast.nio.Bits.CACHE_LINE_LENGTH;
+import static com.hazelcast.nio.Bits.LONG_SIZE_IN_BYTES;
+
+
+public abstract class CallIdSequence {
+
+    /**
+     * Creates the next call-id.
+     * <p/>
+     *
+     * @return the generated callId.
+     */
+    public abstract long next();
+
+    /**
+     * Creates the call id without applying any backpressure
+     *
+     * @return the generated call id.
+     */
+    public abstract long renew();
+
+    public abstract void complete();
+
+    /**
+     * A {@link com.hazelcast.spi.impl.operationservice.impl.CallIdSequence} that provided backpressure by taking
+     * the number in flight operations into account when a call-id needs to be determined.
+     * <p/>
+     * It is possible to temporary exceed the capacity:
+     * - due to system operations
+     * - due to racy nature of checking if space is available and getting the next sequence.
+     * <p/>
+     */
+    public static final class CallIdSequenceWithBackpressureViaException extends CallIdSequence {
+        private static final int INDEX_HEAD = 7;
+        private static final int INDEX_TAIL = 15;
```

Comment:
I know, this part is taken from PR #7075
It would be better that calculating `INDEX_TAIL` as `INDEX_HEAD + CACHE_LINE_LENGTH / LONG_SIZE_IN_BYTES` so if `com.hazelcast.nio.Bits#CACHE_LINE_LENGTH` is changed, `INDEX_TAIL` will be updated accordingly.